maketeams3.py

```diff
@@ -258,60 +313,219 @@ def testSwap(teama, playera, teamb, playerb, board):
     for player in players:
         player.setReqMet()
 
-    #WIP for upload format for heltour
+    return {'teams': teams,
+            'players': players,
+            'alternates': alternates,
+            'team_rating_bounds': team_rating_bounds,
+            'alt_rating_bounds': alt_rating_bounds,
+            'alts_split': alts_split}
+
+
+# Reduce variance functions
+
+def intersection(lst1, lst2):
+    return set(lst1).intersection(set(lst2))
+
+
+# Does this swap have a neutral effect on happiness
+def is_neutral_swap(swap):
+    def count_on_team(attr, player, team):
+        n = len(intersection(getattr(player, attr), team.boards))
+        n += len([p for p in team.boards if player in getattr(p, attr)])
+        return n
+
+    count_friends_on_team = partial(count_on_team, 'friends')
+    count_avoids_on_team = partial(count_on_team, 'avoid')
+
+    pa, pb = swap
+    if count_friends_on_team(pa, pa.team) != count_friends_on_team(pa, pb.team)\
```

Comment:
I don't think the logic is correct here.  pa has 4 friends on their current team and only two friends on the other team, and pb has 2 friends on their current team and 4 friends on the other team, isn't that also a neutral swap? League happiness remains the same with the swap in that case?

Reply:
It might not actually matter in terms of the performance of the algorithm, but from the  name of the function, I would expect the above swap to be neutral as well.

Reply:
I think you're correct. I will look into this.